deploy/LiveDeploy.js

```diff
@@ -0,0 +1,330 @@
+            Balances: libBalances.address,
+            Prices: libPricing.address,
+        },
+        log: true,
+    })
+
+    // deploy Tracer perps factory
+    let factory = await deploy("TracerPerpetualsFactory", {
+        args: [
+            perpsDeployer.address,
+            liquidationDeployer.address,
+            insuranceDeployer.address,
+            pricingDeployer.address,
+            deployer, // governance address
+        ],
+        from: deployer,
+        log: true,
+    })
+
+    let maxLeverage = ethers.utils.parseEther("12.5")
+    let tokenDecimals = new ethers.BigNumber.from("18").toString()
+    let feeRate = 0 // 0 percent
+    let fundingRateSensitivity = 1
+    let maxLiquidationSlippage = ethers.utils.parseEther("50") // 50 percent
+    let deleveragingCliff = ethers.utils.parseEther("20") // 20 percent
+    let lowestMaxLeverage = ethers.utils.parseEther("12.5") // Default -> Doesn't go down
+    let _insurancePoolSwitchStage = ethers.utils.parseEther("1") // Switches mode at 1%
+
+    var deployTracerData = ethers.utils.defaultAbiCoder.encode(
+        [
+            "bytes32", //_marketId,a
+            "address", //_tracerQuoteToken,
+            "uint256", //_tokenDecimals,
+            "address", //_gasPriceOracle,
+            "uint256", //_maxLeverage,
+            "uint256", //_fundingRateSensitivity,
+            "uint256", //_feeRate
+            "address", // _feeReceiver,
+            "uint256", // _deleveragingCliff
+            "uint256", // _lowestMaxLeverage
+            "uint256", // _insurancePoolSwitchStage
+        ],
+        [
+            ethers.utils.formatBytes32String("TEST1/USD"),
+            token.address,
+            tokenDecimals,
+            gasPriceOracleAdapter.address,
+            maxLeverage,
+            fundingRateSensitivity,
+            feeRate,
+            deployer,
+            deleveragingCliff,
+            lowestMaxLeverage,
+            _insurancePoolSwitchStage,
+        ]
+    )
+
+    // this deploys a tracer, insurance, pricing and liquidation contract
+    await deployments.execute(
+        "TracerPerpetualsFactory",
+        {
+            from: deployer,
+            log: true,
+        },
+        "deployTracer",
+        deployTracerData,
+        oracleAdapter.address,
+        maxLiquidationSlippage
+    )
+
+    const tracerInstance = new ethers.Contract(
+        await deployments.read("TracerPerpetualsFactory", "tracersByIndex", 0),
+        tracerAbi
+    ).connect(signers[0])
+
+    let insurance = await tracerInstance.insuranceContract()
+    let pricing = await tracerInstance.pricingContract()
+    let liquidation = await tracerInstance.liquidationContract()
+
+    // Set Trader.sol to be whitelisted, as well as deployer (for testing purposes)
+    await tracerInstance.setWhitelist(trader.address, true)
+    await tracerInstance.setWhitelist(deployer, true)
+
+    // verify
+    await hre.run("verify:verify", {
```

Comment:
might be a silly question, but why don't we just run all of these without `await`, since we don't actually do anything with the results, or need them to be finished before proceeding if you get what I mean.

Reply:
Won't we still need to drive each promise?

Reply:
Yeah good point, I guess I just had the await to see if any failed and run them sequentually. Might hit rate limits on Etherscan? But i think that should be fine.

Reply:
@sporejack @BenjaminPatch had a play around locally, it seems to make sense to just keep these async so its clear when failures occur. It doesn't add too much time to the flow either way. If it gets too annoying we can do them synchronously later on i feel